ironic/drivers/modules/drac/management.py

```diff
@@ -302,14 +313,173 @@ def set_boot_device(node, device, persistent=False):
+                                                job_ids=['JID_CLEARALL'])
+            except sushy.exceptions.SushyError as e:
+                error_msg = ("Sushy OEM extension Python package "
+                          "sushy-oem-idrac failed to clear job queue"
+                          "for %(node)s, Error : %(error)s" %
+                          {'node':task.node.uuid, 'error':e})
+                LOG.error(error_msg)
+                raise exception.RedfishError(error=error_msg)
+
+        if delete_job_response.status_code == _JOB_RESPONSE_CODE:
+            msg = ("Job queue cleared for node %(node)s via OEM" %
+                    {'node': task.node.uuid})
+            LOG.info(msg)
+            LOG.debug(msg)
+        else:
+            error_msg = ("Failed to clear job queue, node : %(node)s " %
+                        {'node': task.node.uuid})
+            LOG.error(error_msg)
+            raise exception.RedfishError(error=error_msg)
+
+        return None
+
+    @METRICS.timer('DracRedfishManagement.reset_idrac')
+    @base.clean_step(priority=0)
+    def reset_idrac(self, task):
+        """Reset the iDRAC.
 
+        :param task: a TaskManager instance containing the node to act on.
+        :returns: None if it is completed.
+        :raises: RedfishError on an error.
+        """
+        system = redfish_utils.get_system(task.node)
+        for manager in system.managers:
+            try:
+                oem_manager = manager.get_oem_extension('Dell')
+            except sushy.exceptions.OEMExtensionNotFoundError as e:
+                error_msg = (_("Search for Sushy OEM extension package "
+                           "'sushy-oem-idrac' failed for node %(node)s. "
+                           "Ensure it is installed. Error: %(error)s") %
+                         {'node': task.node.uuid, 'error': e})
+                LOG.error(error_msg)
+                raise exception.RedfishError(error=error_msg)
+            try:
+                reset_job_response = oem_manager.reset_idrac(task)
+            except sushy.exceptions.SushyError as e:
+                error_msg = ("Sushy OEM extension Python package "
+                          "sushy-oem-idrac failed to reset idrac"
+                          "for %(node)s, Error : %(error)s" %
+                          {'node':task.node.uuid, 'error':e})
+                LOG.error(error_msg)
+                raise exception.RedfishError(error=error_msg)
+
+        redfish_address = task.node.driver_info.get('redfish_address')
+        redfish_address = urlparse(redfish_address)
+        redfish_hostname = redfish_address.netloc
+
+        if reset_job_response.status_code == _JOB_RESPONSE_CODE:
+            msg = ("iDRAC was reset for node %(node)s via OEM,"
+                    "waiting for return to operational state" %
+                    {'node':task.node.uuid})
+            LOG.info(msg)
+            LOG.debug(msg)
```

Comment:
Let's remove the LOG.debug(msg) line above.  No need to log it twice.

Reply:
Done.